app/models/participant_declaration.rb

```diff
@@ -81,6 +81,8 @@ class ParticipantDeclaration < ApplicationRecord
   scope :paid_npqs_for_lead_provider, ->(lead_provider) { paid_for_lead_provider(lead_provider).npq }
   scope :paid_uplift_for_lead_provider, ->(lead_provider) { paid_for_lead_provider(lead_provider).uplift }
 
+  after_create :create_initial_declaration_state
```

Comment:
Do you want this build as part of the create? (e.g., `before_create` and `declaration_states.build(...)`)

Reply:
- i'm not sure `before_create` would work as it needs to reference the persisted record.
- `build` i think could work but where would you stick it? and also what would you set the state to when you call `build` as it could be `submitted` or `ineligible`

Reply:
```
  before_create :build_initial_declaration_state
  
  private
  
  def build_initial_declaration_state
    declaration_states.build(state: state)
  end
```

I _think_ this should work if `inverse_of` is correctly set on each side of the association. The save on the declaration will save the state.

Reply:
Worked without `inverse_of `. Saving the parent seems to also save the child